packages/allow-deny/allow-deny.js

```diff
@@ -138,15 +138,34 @@ CollectionPrototype._defineMutationMethods = function(options) {
           // between arbitrary client-specified _id fields and merely
           // client-controlled-via-randomSeed fields.
           let generatedId = null;
-          if (method === "insert" && !hasOwn.call(args[0], '_id')) {
-            generatedId = self._makeNewID();
+          let isBulkInsert = false;
+          if (method === "insert") {
+            isBulkInsert = Array.isArray(args[0]);
+            if (!isBulkInsert && !hasOwn.call(args[0], '_id')) {
+              generatedId = self._makeNewID();
+            }
+            if (isBulkInsert) {
+              generatedId = [];
+              args[0].forEach(doc => {
+                if (!hasOwn.call(doc, '_id')) {
+                  generatedId.push(self._makeNewID());
+                }
+              });
+            }
           }
 
           if (this.isSimulation) {
             // In a client simulation, you can do any mutation (even with a
             // complex selector).
-            if (generatedId !== null)
-              args[0]._id = generatedId;
+            if (generatedId !== null) {
+              if (!isBulkInsert) {
+                args[0]._id = generatedId;
+              } else {
+                args[0].forEach(doc => {
+                  doc._id = generatedId;
```

Comment:
If I'm reading this correctly, this line is assigning an array to doc._id rather than an individual ID.

Reply:
Good catch! Fixed.

Still having client side failures on async calls...